controllers/common/common.go

```diff
@@ -303,3 +308,41 @@ func MonitoringServiceName(name string) string {
 func LogsPipelineName(namespace, name string) string {
 	return strings.Join([]string{namespace, name, "logs"}, "-")
 }
+
+func GetMetaHTTPServiceURL(cluster *v1alpha1.GreptimeDBCluster) string {
+	return fmt.Sprintf("http://%s.%s:%d", ResourceName(cluster.GetName(), v1alpha1.MetaRoleKind), cluster.GetNamespace(), cluster.Spec.Meta.RPCPort)
+}
+
+// SetMaintenanceMode requests the metasrv to set the maintenance mode.
+func SetMaintenanceMode(metaHTTPServiceURL string, enabled bool) error {
+	requestURL := fmt.Sprintf("%s/admin/maintenance?enable=%v", metaHTTPServiceURL, enabled)
+
+	operation := func() error {
+		rsp, err := http.Get(requestURL)
+		if err != nil {
+			return err
+		}
+		defer rsp.Body.Close()
+
+		if rsp.StatusCode != http.StatusOK {
+			return fmt.Errorf("failed to request metasrv '%s' for set maintenance mode '%v', status code: %d", metaHTTPServiceURL, enabled, rsp.StatusCode)
+		}
```

Comment:
Using the GET method to set maintenance mode is no longer recommended. Let’s use the POST method instead.